examples/cosmos/ics20/denomTrace.qnt

```diff
@@ -88,20 +86,27 @@ module denomTraceTest {
     path: [],
   }
 
-  /// Finds the correct channels for a transfer according to the chain setup
-  /// described above
-  pure def getChannelsForTransfer(sourceChain: str, destChain: str): { sourceChannel: str, destChannel: str } =
-    pure val channels = Map(
-      "A" -> "channelToA",
+  /// Map from chain identifiers to their stored channels, which are maps from
+  /// counterparty chain identifiers to channel identifiers. For example, chain
+  /// A connects to chain B through channel "channelToB".
+  pure val channelTopology = Map(
+    "A" -> Map(
       "B" -> "channelToB",
       "C" -> "channelToC"
+    ),
+    "B" -> Map(
+      "A" -> "channelToA",
+      "C" -> "channelToC"
+    ),
+    "C" -> Map(
+      "A" -> "channelToA",
+      "B" -> "channelToB"
     )
-    ({
-      // If this is a transfer from A -> B, then the source channel is `channelToB`
-      sourceChannel: channels.get(destChain),
-      // If this is a transfer from A -> B, then the dest channel is `channelToA`
-      destChannel: channels.get(sourceChain),
-    })
+  )
+
+  action Transfer(sourceChain, destChain) =
+    val movingBack = movingBackAlongTrace(trace, { port: "transfer", channel: channelTopology.get(sourceChain).get(destChain) })
+    trace' = updateTrace(movingBack, trace, { port: "transfer", channel: channelTopology.get(destChain).get(sourceChain) })
```

Comment:
One thing I repeatedly find confusing is that the same IBC channel is sometimes referred to by its local channel name, and sometimes by its remote name.

Should we introduce a type alias
```bluespec
type Channel = { local: HopElement, remote: HopElement }
```
and a def that constructs this representation from a `ChannelTopology`?
We could then update `movingBackAlongTrace` and `updateTrace` to simply take a `Channel`. The record field names would also make it more obvious which channel end is referred to.

Reply:
I actually find it important to only handle source channel ends on `movingBackAlongTrace` and destination channel ends on `updateTrace` since that is what is happening in fact: the check on `movingBackAlongTrace` compares one side of the hop against the other side of the hop that was registered in the trace (by `updateTrace`). This was something that helped me see things clearer and I'd like to avoid generalizing this parameters (to receive more information than what's actually being used).

Also, what's local and what's remote is not really part of this spec scope IMO. Locality of channel information is not relevant here.

Reply:
Maybe this is something we should follow-up on outside of this PR, to get it merged.

Reply:
But source and destination is an absolute truth, it is not relative to the perspective of each chain, as `local` and `remote` might be.

But I agree, we can revisit and iterate later :smile: will merge.